special-pages/pages/duckplayer/app/components/Player.jsx

```diff
@@ -103,8 +112,16 @@ function useIframeEffects(src) {
             features.titleCapture(),
             features.mouseCapture(),
             features.errorDetection(),
+            features.replaceWatchLinks(),
         ];
 
+        // Handle watch link clicks in iframe
+        window.addEventListener(WATCH_LINK_CLICK_EVENT, () => {
+            if (embed) {
+                openOnYoutube(embed);
+            }
+        });
+
```

Comment:
this is better for sure. if you want to go 1 step further just pass a callback instead, like this https://github.com/duckduckgo/content-scope-scripts/commit/19f6d41f5b270870a9bf59c9b57284b3dfac1cc3

^ no need to apply this, just showing

Reply:
Applied. Thanks for removing the leftovers from previous iterations too.